cmd/zoekt-sourcegraph-indexserver/backoff_test.go

```diff
@@ -0,0 +1,249 @@
+package main
+
+import (
+	"github.com/sourcegraph/log/logtest"
+	"testing"
+	"time"
+)
+
+func TestQueue_BackoffOnFail(t *testing.T) {
+	backoffDuration := 1 * time.Millisecond
+	maxBackoffDuration := backoffDuration * 2
+
+	queue := NewQueue(backoffDuration, maxBackoffDuration, logtest.Scoped(t))
+	opts := IndexOptions{RepoID: 1, Name: "foo"}
+
+	queue.AddOrUpdate(opts)
+
+	// Empty queue
+	for ok := true; ok; _, ok = queue.Pop() {
+	}
```

Comment:
minor: you repeat this pattern in a few tests maybe pull out a helper with a nice name. eg `func emptyQueue(queue) { ... }`

Reply:
updated